mlx-rs/src/ops/indexing/mod.rs

```diff
@@ -320,65 +320,88 @@ impl Array {
     pub fn take_device(
         &self,
         indices: &Array,
-        axis: i32,
+        axis: impl Into<Option<i32>>,
         stream: impl AsRef<Stream>,
     ) -> Result<Array, Exception> {
         unsafe {
-            let c_array = try_catch_c_ptr_expr! {
-                mlx_sys::mlx_take(self.c_array, indices.c_array, axis, stream.as_ref().as_ptr())
+            let c_array = match axis.into() {
+                Some(axis) => try_catch_c_ptr_expr! {
+                    mlx_sys::mlx_take(self.c_array, indices.c_array, axis, stream.as_ref().as_ptr())
+                },
+                None => try_catch_c_ptr_expr! {
```

Comment:
Don't we already have `take_all`? https://github.com/oxideai/mlx-rs/blob/aac7e03e609b20b5b28aae3bb5eef61dcc2319f9/mlx-rs/src/ops/indexing/mod.rs#L341 

I think it was intentional that we separated `take` and `take_all`

Reply:
I think we separated `take` and `take_all` because this pattern also appears in other functions, for example `argsort` vs `argsort_all`, `stack` vs `stack_all`. I'm thinking we probably want to be consistent on this? Either all these functions don't have a `*_all` variants or all of them do?